data-pipeline/src/span_concentrator/aggregation.rs

```diff
@@ -108,32 +110,35 @@ impl<'a> AggregationKey<'a> {
     ///
     /// If `peer_tags_keys` is not empty then the peer tags of the span will be included in the
     /// key.
-    pub(super) fn from_span(span: &'a SpanBytes, peer_tag_keys: &'a [String]) -> Self {
+    pub(super) fn from_span<T>(span: &'a Span<T>, peer_tag_keys: &'a [String]) -> Self
+    where
+        T: SpanText + AsRef<str>,
```

Comment:
Do you need the extra `AsRef<str>` here? SpanText already extends `Borrow<str>` so you should be able to use `s.borrow()` instead of `s.as_ref()`